Goal App/Services/FirestoreService.swift

```diff
@@ -12,23 +12,19 @@ public class FirestoreService {
     public static let shared = FirestoreService()
     private let database = Firestore.firestore()
 
-    public func create(id: String, task: String, createdDate: Date, dueDate: Date) {
+    func create(id: String, task: String, createdDate: Date, dueDate: Date) {
         let documentReference = database.collection("Goals").document(id)
         documentReference.setData(["task": task, "completed": false, "goalCreatedDate": createdDate, "goalDueDate": dueDate])
     }
 
-    internal func getGoals(completionHandler: @escaping (Result<[Goal], Error>) -> Void) {
-        database.collection("Goals").getDocuments { snapshot, error in
-            if error == nil {
-                if let snapshot = snapshot {
-                    let result = snapshot.documents.map { document in
-                        Goal(id: document.documentID, task: document["task"] as? String ?? "", completed: document["completed"] as? Bool ?? false, goalCreatedDate: document["goalCreatedDate"] as? Date ?? Date.now, goalDueDate: document["goalDueDate"] as? Date ?? Date.now)
-                    }
-                    completionHandler(.success(result))
-                }
-            } else {
-                completionHandler(.failure(error!))
+    func getGoals() async -> [Goal]? {
+        do {
+            let snapshot = try await database.collection("Goals").getDocuments()
+            return snapshot.documents.map { document in
+                Goal(id: document.documentID, task: document["task"] as? String ?? "", completed: document["completed"] as? Bool ?? false, goalCreatedDate: document["goalCreatedDate"] as? Date ?? Date.now, goalDueDate: document["goalDueDate"] as? Date ?? Date.now)
             }
+        } catch {
+            return nil
```

Comment:
return an empty array instead of `nil`, 
a function should return something as compared to nothing.
`nil` means nothing and an `empty array` means something